bpf/lib/nodeport.h

```diff
@@ -1110,15 +1110,16 @@ static __always_inline int rev_nodeport_lb6(struct __ctx_buff *ctx, __u32 *ifind
 			if (eth_store_saddr(ctx, fib_params.smac, 0) < 0)
 				return DROP_WRITE_ERROR;
 		}
-	} else {
-		if (!bpf_skip_recirculation(ctx)) {
-			ctx_skip_nodeport_set(ctx);
-			ep_tail_call(ctx, CILIUM_CALL_IPV6_FROM_NETDEV);
-			return DROP_MISSED_TAIL_CALL;
-		}
+
+		return CTX_ACT_REDIRECT;
 	}
 
-	return CTX_ACT_REDIRECT;
+	if (bpf_skip_recirculation(ctx))
+		return DROP_NAT_NO_MAPPING;
```

Comment:
Makes sense to drop given the ifindex 0.. so there's no known call-site where this was set?

Reply:
> Makes sense to drop given the ifindex 0.. so there's no known call-site where this was set?

Not sure I follow. The `from-container` path sets the `skip-recirculation` flag before it tail-calls, and we **should** always find a matching CT entry. So this just handles the "impossible" condition that we don't find a CT entry.